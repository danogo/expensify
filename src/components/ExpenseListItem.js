import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';

const ExpenseListItem = ({description, amount, createdAt, dispatch, id}) => (
  <div>
    <h3>{description}</h3>
    <p>Amount: {amount}</p>
    <p>Created at: {createdAt}</p>
    <button onClick={(e) => {
      dispatch(removeExpense({id}));
    }}>Remove</button>
  </div>
);

// here we connect just to get dispatch method to our WrappedComponent props
export default connect()(ExpenseListItem);