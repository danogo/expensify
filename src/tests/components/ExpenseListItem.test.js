import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import mockExpenses from '../fixtures/expenses';

test('should render ExpenseListItem correctly', () => {
  const wrapper = shallow(<ExpenseListItem {...mockExpenses[2]}/>);
  expect(wrapper).toMatchSnapshot();
});