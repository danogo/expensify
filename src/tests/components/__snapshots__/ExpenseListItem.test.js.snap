// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/3"
  >
    <h3>
      Book
    </h3>
  </Link>
  <p>
    Amount: 
    13370
  </p>
  <p>
    Created at: 
    345600000
  </p>
</div>
`;
